### DISCUSSION TITLE
Unit 8 - advanced foundry course mintNftOnContract function

### DISCUSSION THREAD

**Discussion by r0bchain** [Cyfrin/foundry-full-course-cu#2032] [General]
Hi there, I am trying to understand this piece of code:

```
function mintNftOnContract(address contractAddress) private {
       
       vm.startBroadcast();
       BasicNft(contractAddress).mintNft(PUG_URI);
       vm.stopBroadcast();
    }
```
I checked the constructor in BasicNft.sol, and this constructor has no arguments, how is it possible the function above is passing a parameter "**contractAddress**". 

```
    constructor() ERC721("Dogie", "DOG") {
        s_tokenCounter = 0;
    }

```

**Comment by EngrPips** [comment]
Hello @r0bertinski, This line `BasicNft(contractAddress).mintNft(PUG_URI);` is not creating a new instance of the `BasicNft` contract, but instead, it is making a contract object by wrapping the interface of the `BasicNft` with the address of an initially deployed instance of `BasicNft`. If we want to create a new instance of `BasicNft,` then the code would be `BasicNft nftContract = new BasicNft()`. Please ask questions if anything is confusing.

**Reply by r0bchain** [reply]
Yeah, I got it, thank you.

**Reply by EngrPips** [reply]
Awesome. I am glad you figured it out. Keep crushing it.
